pkg/client/attach.go

```diff
@@ -0,0 +1,257 @@
+package client
+
+import (
+	"context"
+	"io"
+	"net"
+
+	"github.com/containers/conmon-rs/internal/proto"
+	"github.com/containers/podman/v3/libpod/define"
+	"github.com/containers/podman/v3/pkg/kubeutils"
+	"github.com/containers/podman/v3/utils"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	AttachPipeStdin  = 1
+	AttachPipeStdout = 2
+	AttachPipeStderr = 3
+)
+
+type TerminalSize struct {
+	Width, Height uint16
+}
+
+type AttachStreams struct {
+	Stdin                                   io.Reader
+	Stdout, Stderr                          io.WriteCloser
+	AttachStdin, AttachStdout, AttachStderr bool
+}
+
+type AttachConfig struct {
+	ID, SocketPath, ExecSession   string
+	Tty, DetachStdin, Passthrough bool
+	Resize                        chan define.TerminalSize
+	AttachReady                   chan bool
+	Streams                       AttachStreams
+	StartFunc                     func() error
+	DetachKeys                    []byte
```

Comment:
I'd love to see doc comments on those values.

Reply:
done!